### ISSUE TITLE
Incorrect optional flag on Identifier

### ISSUE THREAD

**Issue by vjeux** [eslint/typescript-eslint-parser#289]
**What version of `typescript-eslint-parser` are you using?**
b7220fdbf5a923c64bb6e5bbbfc2ad689ec779eb

**What code were you trying to parse?**

```js
class X {
  private scanner? = undefined;
}
```

**What did you expect to happen?**

`undefined` not having `optional: true`

**What happened?**

```js
{ type: 'Identifier',
  range: [ 47, 56 ],
  loc: { start: { line: 2, column: 23 }, end: { line: 2, column: 32 } },
  name: 'undefined',
  optional: true }
```

this makes prettier print is as

```js
class SemicolonWalker {
  private scanner? = undefined?;
}
```

which isn't valid

**Comment by soda0289** [comment]
Good catch, thanks
